Documents/AnalyticsToolDesignDocument.md

```diff
@@ -0,0 +1,48 @@
+# VFS For Git: Analytics Tooling Design Document
+
+This document details the design of and plan for the **VFS4G analytics tool**. The tool aims to help users identify reasons why their enlistment may be slowing down and the `gvfs` commands running slowly. By collecting and reporting statistics, the power to uncover and hopefully fix problems as they arrive moves to the user, empowering them to use the software to its fullest potential.
+
+## Problem
+
+When users unintentionally hydrate too many files within their enlistment, commands like checkout and commit increase their running time by several orders of magnitude. This causes unacceptable losses in performance and greatly hinders users as they go about their daily development cycle. The primary metric of the overhydration is the number of placeholder files which have been pulled from the remote into the local system. Slowdowns may also stem from having many modified files or directories which increases the amount of work that git does under the hood. Additionally, if a local enlistment has been in use for a long time, the number of loose object may eventually reach the order of several thousand in which case that also contributes to the loss in performance.
+
+## Provided Statistics
+
+### Safe to access

```

Comment:
What does this mean?  Does `access` mean only to read the information from?

Also is this a list of the statistics that you are going to be tracking or the files that will be read?

Reply:
This is a list of the statistics I plan to extract from the enlistment and provide to the user as a whole, and previously it included some of the assets I was going to need to get them as far as I could tell. It has since been updated, after I finish reading through the recent feedback I will recommit the updated document. I have since removed the "safe to access" tag which I had initially meant to contain the statistics where if we accessed them there was no risk of accidentally hydrating files on disk farther than already done (like we talked about last night with downloading file sizes).